libs/core/src/lib/checkbox/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,143 @@
+import {Component, EventEmitter, forwardRef, Input, Output} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+import {FdCheckboxValues} from './fd-checkbox-values.interface';
+
+let checkboxUniqueId: number = 0;
+
+@Component({
+    selector: 'fd-checkbox',
+    templateUrl: './checkbox.component.html',
+    styleUrls: ['./checkbox.component.scss'],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => CheckboxComponent),
+            multi: true
+        }
+    ]
+})
+export class CheckboxComponent implements ControlValueAccessor {
+
+    /** Sets [id] property of input, binds input with input label using [for] property. */
+    @Input()
+    inputId: string = `fd-checkbox-${checkboxUniqueId++}`;
+
+    /** State of control, changes visual appearance of control. */
+    @Input()
+    state: 'valid' | 'invalid' | 'info' | 'warning';
+
+    /** Sets [name] property of input. */
+    @Input()
+    name: string;
+
+    /** Sets text of control label. */
+    @Input()
+    label: string;
+
+    /** Allows to disable/enable control. */
+    @Input()
+    disabled: boolean;
+
+    /** Allows to minimize control to compact mode. */
+    @Input()
+    compact: boolean;
+
+    /** Enables controls third state. */
+    @Input()
+    tristate: boolean;
+
+    /** Allows to prevent user from manually selecting controls third state. */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** Values returned by control. */
+    @Input()
+    values: FdCheckboxValues = {trueValue: true, falseValue: false, thirdStateValue: null};
```

Comment:
I know by doing this we are giving them flexibility but shouldn't a checkbox be strictly a boolean ?

Reply:
Checkbox control returns boolean values by default.
I think giving user possibility to define values returned by checkbox is a nice and useful feature.